In this assignment, we aim to answer three questions related to U.S. airports and weather conditions using Python and pandas for data analysis. The questions focus on identifying the northernmost and easternmost airports in the United States, as well as determining the windiest airport in the New York area on a specific date.

We will use the following datasets for our analysis:

Airports Dataset: Contains information about airports in the United States, including their geographic coordinates (latitude and longitude), which will help us determine the northernmost and easternmost airports. 

Weather Dataset: Includes weather observations, such as wind speed, for New York area airports. This will allow us to identify the windiest airport on February 12th, 2013.

Flights Dataset: Although not directly needed for this specific task, the flights data could help in cross-referencing airport codes and handling additional airport information if necessary.

We will employ Python and the pandas library to load, filter, and analyze these datasets, using external research to validate our findings where needed. Throughout this process, we will document any assumptions and data adjustments made to ensure the accuracy and validity of our results.

In [3]:
import pandas as pd

In [11]:
# Load the airports data
airports = pd.read_csv('C:\\Users\\danie\\Downloads\\airports.csv')

In [19]:
airports.head()

faa                           name        lat        lon   alt  tz dst  \
0  04G              Lansdowne Airport  41.130472 -80.619583  1044  -5   A   
1  06A  Moton Field Municipal Airport  32.460572 -85.680028   264  -6   A   
2  06C            Schaumburg Regional  41.989341 -88.101243   801  -6   A   
3  06N                Randall Airport  41.431912 -74.391561   523  -5   A   
4  09J          Jekyll Island Airport  31.074472 -81.427778    11  -5   A   

              tzone  
0  America/New_York  
1   America/Chicago  
2   America/Chicago  
3  America/New_York  
4  America/New_York

In [21]:
northernmost_airports = airports.sort_values(by='lat', ascending=False).head(5)
northernmost_airports

faa                                        name        lat         lon  \
417  EEN                     Dillant Hopkins Airport  72.270833   42.898333   
230  BRW                  Wiley Post Will Rogers Mem  71.285446 -156.766003   
110  AIN                          Wainwright Airport  70.638056 -159.994722   
708  K03                               Wainwright As  70.613378 -159.860350   
152  ATK  Atqasuk Edward Burnell Sr Memorial Airport  70.467300 -157.436000   

     alt  tz dst              tzone  
417  149  -5   A                NaN  
230   44  -9   A  America/Anchorage  
110   41  -9   A  America/Anchorage  
708   35  -9   A  America/Anchorage  
152   96  -9   A  America/Anchorage

- What is the northernmost airport in the United States?

After analyzing the airports dataset, the northernmost airport in the United States is Dillant Hopkins Airport, located at latitude 72.270833.

- Some assumptions are:

The dataset used contains accurate and up-to-date geographic coordinates for all U.S. airports.

The latitude values in the dataset were used to determine how far north an airport is, with higher positive values indicating a location further north.

In [26]:
# Sort by longitude to get the easternmost airports
easternmost_airports = airports.sort_values(by='lon', ascending=True).head(5)
easternmost_airports

faa              name        lat         lon  alt  tz dst  \
95    ADK      Adak Airport  51.878000 -176.646000   18  -9   A   
112   AKB      Atka Airport  52.220278 -174.206389   56  -9   A   
522   GAM   Gambell Airport  63.766800 -171.733000   27  -9   A   
1281  SVA  Savoonga Airport  63.686400 -170.493000   53  -9   A   
1251  SNP    St Paul Island  57.167333 -170.220444   63  -9   A   

                  tzone  
95    America/Anchorage  
112   America/Anchorage  
522   America/Anchorage  
1281  America/Anchorage  
1251  America/Anchorage

- What is the easternmost airport in the United States?

After analyzing the airports dataset, the easternmost airport in the United States is Adak Airport, located at longitude -176.646000.

- Some assumptions are:

Longitude values were used to determine how far east an airport is, with lower (more negative) longitude values indicating locations further east (since the U.S. is located west of the Prime Meridian).

External validation confirmed that Adak Airport is recognized as the easternmost U.S. airport based on its longitude.

In [29]:
# Load the weather data
weather = pd.read_csv("C:\\Users\\danie\\Downloads\\weather.csv")

In [31]:
feb_12_weather = weather[(weather['year'] == 2013) & 
                         (weather['month'] == 2) & 
                         (weather['day'] == 12)]

In [33]:
windiest_airport = feb_12_weather.groupby('origin')['wind_speed'].max().sort_values(ascending=False)
windiest_airport

origin
EWR    1048.36058
LGA      23.01560
JFK      20.71404
Name: wind_speed, dtype: float64

- On February 12th, 2013, which New York area airport had the windiest weather?

After analyzing the weather data for New York area airports (JFK, LGA, EWR), the airport with the highest recorded wind speed on February 12th, 2013, was Newark Liberty International Airport (EWR) , with a wind speed of 1048.3605 mph.

- Some assumptions are:

The wind speed values in the dataset reflect the maximum wind speed recorded during the day.

Any differences in wind conditions throughout the day were averaged out, and the peak wind speed was used to determine the result.
